azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/GenericResource.java

```diff
@@ -140,10 +159,62 @@
     }
 
     /**
+     * A generic resource update allowing to change the resource properties.
+     */
+    interface UpdateWithProperties {
+        /**
+         * Specifies other properties of the resource.
+         *
+         * @param properties the properties object
+         * @return the next stage of generic resource update
+         */
+        Update withProperties(Object properties);
+    }
+
+    /**
+     * A generic resource update allowing to change the parent resource.
+     */
+    interface UpdateWithParentResource {
+        /**
+         * Specifies the parent resource.
+         *
+         * @param parentResourceId the UUID of the parent resource
```

Comment:
is this UUID or general azure resource id of the form `subscriptions/<subscription-id>/resourceGroups/<rg-name>/..`?